drivers/io1_xplained/io1_xplained.c

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2016 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_io1_xplained
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the Atmel IO1 Xplained extension
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <math.h>
+
+#include "log.h"
+#include "io1_xplained.h"
+#include "io1_xplained_internals.h"
+#include "io1_xplained_params.h"
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG        (1)
+#include "debug.h"
+
+/**
+ * @brief   Allocation of memory for device descriptors
+ */
+io1_xplained_t io1_xplained_devs[IO1_XPLAINED_NUMOF];
+
+/*---------------------------------------------------------------------------*
+ *                          IO1 Xplained Core API                            *
+ *---------------------------------------------------------------------------*/
+
+int io1_xplained_init(io1_xplained_t *dev, i2c_t i2c, uint8_t addr)
+{
+    dev->i2c_dev = i2c;
+    dev->addr = (TEMPERATURE_BASE_ADDR | addr);
+
+    /* Initialize I2C interface */
+    if (i2c_init_master(dev->i2c_dev, I2C_SPEED_NORMAL)) {
+        DEBUG("[Error] I2C device not enabled\n");
+        return -1;
+    }
+
+    if (gpio_init(IO1_LED_PIN, GPIO_OUT) < 0) {
+        DEBUG("[Error] GPIO LED not enabled\n");
+        return -1;
+    }
+    
+    DEBUG("[Info] IO1 Xplained extension initialized!\n");
+    DEBUG("[Info] address: %d\n", (int)dev->addr);
+
+    return 0;
+}
+
+void io1_xplained_auto_init(void)
+{
+    for (unsigned i = 0; i < IO1_XPLAINED_NUMOF; i++) {
+        if (io1_xplained_init(&io1_xplained_devs[i],
+                              io1_xplained_params[i].i2c_dev,
+                              io1_xplained_params[i].addr) < 0) {
+            LOG_ERROR("Unable to initialize IO1 Xplained #%i\n", i);
+        }
+#ifdef MODULE_SAUL_REG
+        for (int j = 0; j < 2; j++) {
+            io1_xplained_saul_reg[i][j].dev = &io1_xplained_devs[i];
+            saul_reg_add(&io1_xplained_saul_reg[i][j]);
+        }
+#endif
+    }
+}
+
+int io1_xplained_read_temperature(io1_xplained_t *dev, int32_t *temperature)
```

Comment:
Well, it wouldn't hurt and speed up getting going with the peripherals. Back then, I had also implemented the SAUL interface but never created a PR since it was only hacked together quickly.